hnn_core/cell_response.py

```diff
@@ -476,3 +505,37 @@ def read_spikes(fname, gid_ranges=None):
         cell_response.update_types(gid_ranges)
 
     return cell_response
+
+
+def read_spikes_hdf5(fname, gid_ranges=None):
```

Comment:
don't create a new public function ... this should be a private function and you should triage the right function from the top level function based on the extension